src/build/stable/commands/compile/javascript.ts

```diff
@@ -25,24 +25,38 @@ function getPrelude(main: string): string {
 
 export function handleClassApiCanister(): string {
     return /*TS*/ `
-        if (globalThis._azleNodeWasmEnvironment === false) {
-            const canisterClassInstance = new Canister.default();
-            globalThis._azleCanisterClassInstance = canisterClassInstance;
-        }
+        const exportedCanisterClassInstance = getExportedCanisterClassInstance();
 
-        const canisterIdlType = IDL.Service(globalThis._azleCanisterMethodIdlTypes);
+        const canisterIdlType = IDL.Service(exportedCanisterClassInstance._azleCanisterMethodIdlTypes);
         const candid = canisterIdlType.accept(new DidVisitor(), {
             ...getDefaultVisitorData(),
             isFirstService: true,
-            systemFuncs: globalThis._azleInitAndPostUpgradeIdlTypes
+            systemFuncs: exportedCanisterClassInstance._azleInitAndPostUpgradeIdlTypes
         });
 
         globalThis._azleGetCandidAndMethodMeta = () => {
             return JSON.stringify({
                 candid: toDidString(candid),
-                methodMeta: globalThis._azleMethodMeta
+                methodMeta: exportedCanisterClassInstance._azleMethodMeta
             });
         };
+        
+        function getExportedCanisterClassInstance() {     
+            try {
+                new Canister.default();
```

Comment:
And the reason we need to initialize the class every time even in the node environment is because even if the decorators always do their thing it's not until the class is initialized that the values actually get set in the globals. Is that right?